### PR TITLE
feat: add basic query-mode system

### PR DESCRIPTION
**Pull request by zakuciael**
## TL;DR
This pull request improves the user experience by allowing users to filter the list of recent projects by an IDE.

---

## What changed
- The `projects` field in the `Mode` struct has been split into `found_projects` and `entries`. `found_projects` stores all the projects found by the matchers, while `entries` stores the projects that match the current user query.
- The `query` field has been added to the `Mode` struct to store the current `IDE` query.
- The `react` method has been updated to handle `Event::CustomInput` events. When such an event is received, the method checks if there is a current `IDE` query. If there is none or if the `alt` flag is set, it attempts to parse the first word of the input as an `IDE` code. If successful, it sets the `query` field and filters the `entries` to only include projects that match the `IDE`. If unsuccessful, it logs a debug message and reloads the mode.
- The `matches` method has been updated to match against the `name` field of the `entries` instead of the `projects`.
- The `preprocess_input` method has been removed as it is no longer needed.

### CODE REVIEW

**Comment by zakuciael** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/zakuciael/rofi-jetbrains/1" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/zakuciael/rofi-jetbrains/2" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/zakuciael/rofi-jetbrains/2?utm_source=stack-comment).
